lib/std/unicode.zig

```diff
@@ -631,3 +631,51 @@ test "utf8ToUtf16LeWithNull" {
         testing.expect(utf16[2] == 0);
     }
 }
+
+pub fn utf8ToUtf16LeStringLiteral(comptime utf8: [:0]const u8) [:0]const u16 {
```

Comment:
The input parameter type need not be null terminated (it should be `[]const u8`). The output parameter type should be `*[N:0]const u16`.

Reply:
I pushed the changes I made in the gist that I linked on IRC a few days ago to this branch. 

As I mentioned there, I wasn't thrilled with how the calculation of the length was being done (once in the return type, once in the body, and once in the actual conversion). The calculate function is the conversion function, minus the actual conversion, so I added an assert to make sure they don't get out of sync.